src/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import {
+  autoUpdate,
+  useFloating,
+  useInteractions,
+  useHover,
+  useFocus,
+  useRole,
+  useDismiss,
+  useId,
+} from '@floating-ui/react-dom-interactions';
+import {TooltipProps} from './types';
+import {withOwnTheme} from '../utils/with-own-theme';
+import {StyledTooltip, StyledTooltipPanel} from './styled';
+import defaults from './defaults';
+import stylePresets from './style-presets';
+import {useControlled} from '../utils/hooks';
+
+const ThemelessTooltip: React.FC<TooltipProps> = ({
+  children,
+  title,
+  placement = 'top',
+  trigger = 'hover',
+  open: openProp,
+  defaultOpen,
+  labelTooltip,
+  overrides,
+  ...props
+}) => {
+  const [open, onOpenChange] = useControlled({
+    controlledValue: openProp,
+    defaultValue: Boolean(defaultOpen),
+  });
+
+  const {x, y, reference, floating, strategy, context} = useFloating({
+    placement,
+    open,
+    onOpenChange,
+    whileElementsMounted: autoUpdate,
+  });
+
+  const {getReferenceProps, getFloatingProps} = useInteractions([
+    useHover(context, {
+      enabled: trigger.includes('hover'),
+    }),
+    useFocus(context, {enabled: trigger.includes('focus')}),
+    useRole(context, {enabled: !labelTooltip, role: 'tooltip'}),
+    useDismiss(context),
+  ]);
+
+  // If tooltip is used as a label, add aria-labelledby to childrenProps and id to StyledTooltip;
+  // Because of above, 'aria-describedby' has different id for reference and floating, hence manually set below as well;
+  const id = useId();
+
+  const labelOrDescProps = {} as {
+    'aria-labelledby': string | null;
+    'aria-describedby': string | null;
+  };
```

Comment:
as far I can see you never have both properties at the same time
```suggestion
  const labelOrDescProps = {} as {
    'aria-labelledby'?: string | null;
    'aria-describedby'?: string | null;
  };
```